src/test/resources/simulators/avdManager.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as Q from "q";
+import {DeviceHelper} from "./deviceHelper";
+
+export interface IAVDManager {
+    // Intends to simulate: android create avd -n <name> -t <targetID> [-<option> <value>] ...
+    create(avdId: string): Q.Promise<void>;
+    // Intends to simulate: emulator -avd WVGA800 -scale 96dpi -dpi-device 160
+    launch(avdId: string): Q.Promise<string>;
+    createAndLaunch(avdId: string): Q.Promise<string>;
+    createAndLaunchAll(...avdIds: string[]): Q.Promise<string[]>;
+}
+
+interface IDeviceState {
```

Comment:
At the moment we don't need any state, but we'll need some in the future when we add more features to this class. I wanted to use this to leave the devices as a dictionary {deviceId: {}, otherDeviceId: {}} rather than use an array [deviceId, otherDeviceId] which we'll need to change to a dictionary when we improve this.